transport/http/outbound.go

```diff
@@ -111,30 +115,63 @@ func (o outbound) Call(ctx context.Context, req *transport.Request) (*transport.
 	deadline, _ := ctx.Deadline()
 	ttl := deadline.Sub(start)
 
-	request, err := http.NewRequest("POST", o.URL, req.Body)
+	req, err := http.NewRequest("POST", o.URL, treq.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	request.Header = applicationHeaders.ToHTTPHeaders(req.Headers, nil)
+	// Apply HTTP Context headers for tracing and baggage carried by tracing.
+	var parent opentracing.SpanContext // ok to be nil
+	if parentSpan := opentracing.SpanFromContext(ctx); parentSpan != nil {
+		parent = parentSpan.Context()
+	}
+	span := o.Tracer.StartSpan(
+		treq.Procedure,
+		opentracing.StartTime(start),
+		opentracing.ChildOf(parent),
+		opentracing.Tags{
+			"yarpc.caller":   treq.Caller,
+			"yarpc.service":  treq.Service,
+			"yarpc.encoding": treq.Encoding,
+		},
+	)
+	ext.PeerService.Set(span, treq.Service)
+	ext.SpanKindRPCClient.Set(span)
+	ext.HTTPUrl.Set(span, req.URL.String())
+	defer span.Finish()
+	ctx = opentracing.ContextWithSpan(ctx, span)
+
+	o.Tracer.Inject(
+		span.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(req.Header),
+	)
+
+	req.Header = applicationHeaders.ToHTTPHeaders(treq.Headers, nil)
 	if hs := baggage.FromContext(ctx); hs.Len() > 0 {
-		request.Header = baggageHeaders.ToHTTPHeaders(hs, request.Header)
+		req.Header = baggageHeaders.ToHTTPHeaders(hs, req.Header)
 	}
 
-	request.Header.Set(CallerHeader, req.Caller)
-	request.Header.Set(ServiceHeader, req.Service)
-	request.Header.Set(ProcedureHeader, req.Procedure)
-	request.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
+	req.Header.Set(CallerHeader, treq.Caller)
+	req.Header.Set(ServiceHeader, treq.Service)
+	req.Header.Set(ProcedureHeader, treq.Procedure)
+	req.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
 
-	encoding := string(req.Encoding)
+	encoding := string(treq.Encoding)
 	if encoding != "" {
-		request.Header.Set(EncodingHeader, encoding)
+		req.Header.Set(EncodingHeader, encoding)
+	}
+
+	response, err := ctxhttp.Do(ctx, o.Client, req)
+
+	if response != nil {
+		span.SetTag("http.status_code", response.StatusCode)
 	}
 
-	response, err := ctxhttp.Do(ctx, o.Client, request)
 	if err != nil {
+		span.SetTag("error", err.Error())
```

Comment:
similar to comment above, you'll want `error` tag to be simply `true`, and log the error message with `logEvent()`